charts/hami/templates/device-plugin/daemonsetvalidator-configmap.yaml

```diff
@@ -0,0 +1,48 @@
+{{- if .Values.devicePlugin.autoValidatorDriver }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: driver-validator-configmap
+data:
+  validator.sh: |
+    #!/bin/zsh
+    if [ -f "/host/usr/bin/nvidia-smi" ]; then
+        driverRoot="/host"
+        isHostDriver="true"
+    else
+        driverRoot="/run/nvidia/driver"
+        isHostDriver="false"
+    fi
+
+    # check this directory /run/nvidia/driver does exist?
+    if [ "$isHostDriver" = "false" ]; then
+      if [ ! -d "/run/nvidia/driver" ]; then
+        echo current node $NODE_NAME not GPU device; sleep infinity
+      fi
+    fi
+
+    sleep_interval=5
+    if [ "$isHostDriver" = "false" ]; then
+        while true; do
+            stat /run/nvidia/validations/.driver-ctr-ready
+            if [ $? -ne 0 ]; then
+              kubectl label node $NODE_NAME hami.io/nvidia-selector="off" --overwrite
+              sleep $sleep_interval
+            else
+              break
+            fi
+        done
+    fi
+    while true; do
+        chroot "$driverRoot" nvidia-smi
+        if [ $? -eq 0 ]; then
+            sleep 30
+            kubectl label node $NODE_NAME hami.io/nvidia-selector="on" --overwrite
```

Comment:
Can you remind me when this label `hami.io/nvidia-selector="on"` will be removed?

Reply:
Current this label can't remove, only change `hami.io/nvidia-selector="off"`, maybe we can remove this label. `kubectl label node $NODE_NAME hami.io/nvidia-selector="off" --overwrite`

Reply:
If the `driver-validator` Pod starts before the `NVIDIA driver`, will this node be labeled as `hami.io/nvidia-selector="off"`?

Reply:
yes.

Reply:
Once the `NVIDIA driver` starts up normally, will the `driver-validator` continue to run?

Reply:
yes, because we should consider NODE restart after, this pod sholud be restart.